compiler/modules/front.syntax/parseStmt.bal

```diff
@@ -116,15 +113,31 @@ function parseStmt(Tokenizer tok) returns Stmt|err:Syntax {
                 return parseMethodCallStmt(tok);
             }
         }
-        [DECIMAL_NUMBER, _]|[STRING_LITERAL, _]|"true"|"false" => {
-            return parseMethodCallStmt(tok);
+        [DECIMAL_NUMBER, _]
+        | [STRING_LITERAL, _]
+        | "true"
+        | "false"
+        | [HEX_INT_LITERAL, _]
+        | [DECIMAL_FP_NUMBER, _, _] => {
+            var peeked = tok.peek();
+            if peeked == "." || (peeked == "[" && !check savePreparseRestore(tok, preparseArrayTypeDesc)) {
+                return parseMethodCallStmt(tok);
+            }
+            return parseVarDeclStmt(tok, startPos);
         }
     }
     return parseError(tok, "unhandled statement");
 }
 
 type AssignOp "="|CompoundAssignOp;
 
+function savePreparseRestore(Tokenizer tok, PreparseFunc func) returns boolean|err:Syntax {
```

Comment:
This function and PreparseFunc should both be in the same file (either here or preparse.bal).